x-pack/plugins/cloud/server/collectors/cloud_usage_collector.ts

```diff
@@ -32,7 +32,7 @@ interface CloudUsage {
 }
 
 export function createCloudUsageCollector(
-  usageCollection: UsageCollectionSetup,
+  usageCollection: ICollectorSet,
```

Comment:
nit: I think this is still `UsageCollectionSetup`. It's getting the entire plugin contract.
Prob what triggered is the name `ICollectorSet`. It sounds a bit confusing for a _CollectorSet_ to be needed to create a usage collector... 

WDYT?

Reply:
I thought about that for a while, and I think the root problem is "naming is hard".
IMO it would make more sense to name it `CollectorManager` instead of `ICollectorSet`.
Then, `UsageCollectionSetup` would have both the `CollectorManager & UsageCountersServiceSetup`.
NB the code above does not need anything about the `UsageCounters`.

The problem is that lots of plugins are already using the global `UsageCollectionSetup`, so changing all the references to the more specific `CollectorManager` would take some work + codeowners approvals.
I only changed this one cause it falls under our ownership.

I can rollback these changes and we can tackle this on a separate PR.